src/nvim/getchar.c

```diff
@@ -2011,9 +2011,7 @@ static int look_in_typebuf(int *mapdepthp, int *keylenp,
   return -1;
 }
 
-// Returns 0 to continue, 1 to break, -1 to carry on in the loop.
-// If it's found a key, leaves the new value of "c" in "*cp"
-static int8_t get_key_from_user(int *cp, int *timedoutp, int *mode_deletedp,
+static int ins_esc_special_case(int *new_wcolp, int *new_wrowp, int *mode_deletedp,
```

Comment:
What I'm trying to say is, you made all this effort to understand this function and make it simpler, but it's still not simple enough to be obvious to the reader. It needs to be consumed in byte sized chunks, and when I see a function with a comment, I'm less tempted to actually go read it and can continue on with the current state in my head.